### ISSUE TITLE
[Transforms] Fail to emit comment that is between class-name and "{" in class declaration when down-level 

### ISSUE THREAD

**Issue by mhegazy** [Microsoft/TypeScript#7914]
**Expected**:

```js
(function (Errors) {
    var ConnectionError /* extends Error */ = (function () {
        function ConnectionError /* extends Error */(request) {
        }
        return ConnectionError /* extends Error */;
    }());
    Errors.ConnectionError /* extends Error */ = ConnectionError /* extends Error */;
})(Errors || (Errors = {}));
```

**Actual**:

```js
(function (Errors) {
    var ConnectionError = (function () {
        function ConnectionError(request) {
        }
        return ConnectionError;
    }());
    Errors.ConnectionError = ConnectionError;
})(Errors || (Errors = {}));

```

Failed tests:
*  tests\cases\compiler\variableDeclaratorResolvedDuringContextualTyping.ts

**Comment by rbuckton** [comment]
@mhegazy I think the expected emit is wrong too. It should be:

```js
(function (Errors) {
    var ConnectionError /* extends Error */ = (function () {
        function ConnectionError(request) {
        }
        return ConnectionError;
    }());
    Errors.ConnectionError = ConnectionError;
})(Errors || (Errors = {}));
```

**Comment by rbuckton** [comment]
The new comment emitter only allows a comment range to be used once in the emit, so there is no way to get the expected output in the original description. My proposed emit above is the only possibility.

@mhegazy Do you know if there are any other tests failing with this specific issue?

**Comment by mhegazy** [comment]
> The new comment emitter only allows a comment range to be used once in the emit, so there is no way to get the expected output in the original description. My proposed emit above is the only possibility.

I agree the original output was wrong. we do not need to match it, we just need to make sure that the comment is emitted once.

> @mhegazy Do you know if there are any other tests failing with this specific issue?

no.

**Comment by rbuckton** [comment]
Fixed in #7940.
